dms/models/dms_file.py

```diff
@@ -588,6 +593,15 @@ def check_directory_access(self, operation, vals=False, raise_exception=False):
     # Constrains
     # ----------------------------------------------------------
 
+    @api.constrains("storage_id", "res_model")
+    def _check_storage_id_attachment_res_model(self):
+        for record in self:
+            if record.storage_id.save_type == "attachment" and not record.res_model:
```

Comment:
If I'm not wrong, a file with this storage also requires a `res_id`, right?

```suggestion
            if record.storage_id.save_type == "attachment" and not (record.res_model and record.res_id):
```